src/formatDate.js

```diff
@@ -8,7 +8,35 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const fromSeparator = fromFormat[3];
+  const toSeparator = toFormat[3];
+  const dateParts = {};
+  const dateValues = date.split(fromSeparator);
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
+    dateParts[fromFormat[i]] = dateValues[i];
+  }
```

Comment:
The loop for assigning date parts to the `dateParts` object is clear, but you might want to check that `dateValues` has the same number of elements as expected by `fromFormat` before the assignment. Even though input validation is not needed as per the task, ensuring the correct number of elements can prevent runtime errors if the function is used in a different context in the future.